sale_product_multi_add/wizards/sale_import_products.py

```diff
@@ -57,8 +57,8 @@ def select_products(self):
                 for item in wizard.items:
                     vals = self._get_line_values(sale, item)
                     if vals:
-                        self.env["sale.order.line"].create(vals)
-
+                        line = self.env["sale.order.line"].create(vals)
+                        line._onchange_discount()
```

Comment:
@st-anagpal You should call it in _get_line_values() after product_id_change().

IMHO, this process should be improved in order to be compliant with further modules on top of this (and that are not aware of).

A solution could be onchange_helper https://github.com/OCA/server-tools/tree/14.0/onchange_helper

Reply:
In theory we should never call onchanges explicitly. A better approach is to use a tool like `onchange_helper` as Denis reported. 

However, as this is a patch and the method `_get_line_values` already calls an onchange, to me it's fair if add this call. @st-anagpal please move it to `_get_line_values`.

It would also be nice to add a specific check for this issue in the tests so that we reduce the chances to have the same issue in the future.